### ISSUE TITLE
Found unparsable section: 'CREATE INDEX' when index involves SQL function

### ISSUE THREAD

**Issue by mtlynch** [sqlfluff/sqlfluff#3237]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

*First: Thanks for making this great tool!*

SQLfluff seems to not recognize correct SQL for creating an index in SQLite if the index involves a SQL function call, in my case `LENGTH`.

### Expected Behaviour

Linting a `CREATE INDEX` query that uses the `LENGTH` function should succeed, as it's valid SQLite:

```bash
$ echo 'CREATE INDEX li1 ON entries_data(id, LENGTH(chunk))' | sqlfluff lint --dialect sqlite -
All Finished 📜 🎉!                     
```

### Observed Behaviour

If I pass in a `CREATE INDEX` SQLite statement that involves a function (`length()`) as part of the index, sqlfluff rejects it as unparseable:

```bash
$ echo 'CREATE INDEX li1 ON entries_data(id, LENGTH(chunk))' | sqlfluff lint --dialect sqlite -
== [stdin] FAIL                                                                                                                                                                                      
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | INDEX li1 ON entries_data(id, LEN...'
```

But the same query without `length()` works fine:

```bash
$ echo 'CREATE INDEX li1 ON entries_data(id, chunk)' | sqlfluff lint --dialect sqlite -
All Finished 📜 🎉!                     
```

### How to reproduce

```bash
$ echo 'CREATE INDEX li1 ON entries_data(id, LENGTH(chunk))' | sqlfluff lint --dialect sqlite -
```

### Dialect

SQLite

### Version

```bash
$ sqlfluff version
0.13.0
$ python --version
Python 3.9.12
```

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mtlynch** [comment]
Thanks so much, @tunetheweb!
